nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+Going forward, we will be diving into how we can use Express in various ways, such as to create a full-stack application using the Model View Controller (MVC) pattern, as well as to create a REST API just like ones you will have used before for things such as the Weather App or the React Shopping Cart projects. Along the way, we will be learning about things like databases, authentication, and much more. There is a lot to take in, so take it steady, and do not be afraid to ask our community for help in [our Discord server](https://discord.gg/theodinproject)!
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Describe what Express is.
+- Describe the relationship between Express and Node.
+- Set up a basic Express server.
+- Describe how an incoming request is processed and ultimately responded to by the server.
+- Describe what a middleware function is.
+- Describe how to use `nodemon` to automatically restart your application when a change is made.
+
+### Setting up Express
+
+Let's make a basic Express app. Inside a new directory, start by running `npm init -y` to create a `package.json`. Once that's created, we can install the Express dependency.
+
+```bash
+npm install express
+```
+
+We can now create an `app.js` file that will serve as the starting point for our Express server. We could name this file anything we want, such as `index.js` or `main.js`, but we will use the same name used in the Express documentation. Inside `app.js`, we will have the following:
+
+```javascript
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("Hello, world!"));
+
+const PORT = 3000;
+app.listen(PORT, () => console.log(`My first Express app - listening on port ${PORT}!`));
+```
+
+Let's break this down. We import `express` then call it to initialize the `app` variable. This is our server.
+
+We then have a `route handler` - the line beginning with `app.get`. We will come back to this in a moment.
+
+Finally, we tell our server to listen for incoming requests requests on port 3000 of our [localhost](https://simple.wikipedia.org/wiki/Localhost) (which is basically just the computer's local connection). While port 3000 is the default choice, you can use any unused port (for example, Vite's dev server uses port 5173 by default). Back in your terminal, if you run `node app.js` then all being well, you should see `My first Express app - listening on port 3000!` logged. Congratulations! Your first Express server is now running.
```

Comment:
Personal preference: We make the congratulations sentence its own para, your call though
```suggestion
Finally, we tell our server to listen for incoming requests requests on port 3000 of our [localhost](https://simple.wikipedia.org/wiki/Localhost) (which is basically just the computer's local connection). While port 3000 is the default choice, you can use any unused port (for example, Vite's dev server uses port 5173 by default). Back in your terminal, if you run `node app.js` then all being well, you should see `My first Express app - listening on port 3000!` logged.

Congratulations! Your first Express server is now running.
```